### PR TITLE
Reposition content

### PR DESCRIPTION
**Pull request by XavierXinweiWang**
Fix the MouseOverListener;
Add the Content Browser Panel on the right side of the window.
Design and implement three button for add note/picture and close window.
Able to add more than 1 sticky note and more than 1 pictures to content right now.

You can only view/edit content for only one node at a time. If you want to view/edit another note, you need to close the content window first.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -10,8 +10,13 @@
...
 </head>
 <body __wid="TDi4iuCp">
-<div id="d3_container" __wid="iZUBjGJV">
-    <svg class="canvas" id="canvas" __wid="PDgACC8e"></svg>
-</div>
+    <div id="CogCanvas">
+        <div id="d3_container" __wid="iZUBjGJV">
+            <svg class="canvas" id="canvas" __wid="PDgACC8e"></svg>
+        </div>
+        <div id="content_container">
```
Comment (stezh) [inline index.html]: It's better to use JS to generate this div when it's initialized. I'll change the canvas correspondingly.

File: index.html [hunk 2]
```diff
@@ -10,8 +10,13 @@
     <link rel="stylesheet" type="text/css" href="styles/main.css">
 </head>
 <body __wid="TDi4iuCp">
-<div id="d3_container" __wid="iZUBjGJV">
-    <svg class="canvas" id="canvas" __wid="PDgACC8e"></svg>
-</div>
+    <div id="CogCanvas">
```
Comment (stezh) [inline index.html]: I think "svg" canvas and "content_container" can belong to a same flex box. Do we still need "div" CogCanvas.

File: src/main.js [hunk 3]
```diff
@@ -699,14 +700,64 @@ function getNodePosition(node){
...
 function addEleContent(e) {
-  var newNodeAddress = "http://webstrates.ucsd.edu/" + hoveredEle;
+  let newNodeAddress = "http://webstrates.ucsd.edu/" + hoveredEle;
+  let wrapper = document.createElement('div');
+  wrapper.setAttribute("id", "addContentWrapper");
+  
+  let toFrame = '<iframe id="addWindow" src="'+ newNodeAddress + '" style="width: 95%; height: 350px;"><p>ERROR: Your browser does not support iframes.</p></iframe>';
+  let addNoteButton = '<button type="button" id="addNoteBtn" onclick="appendNote()">Add Sticky Note</button> ';
+  let addPicButton = '<input type="file" id="addPicBtn" onchange="appendPic()"> ';
+  let closeButton = '<button type="button" id="closeWindowBtn" onclick="closeContentWindow()">Close Content Window</button> ';
```
Comment (stezh) [inline src/main.js]: It is better to use d3 or other JS function to insert new element to the DOM.

File: src/main.js [hunk 4]
```diff
@@ -699,14 +700,64 @@ function getNodePosition(node){
...
+
+  addEleToList(e);
+    
+  addWindowOpen = true;
+
+}
+
+function closeContentWindow() {
+  let wrapper = document.getElementById("addContentWrapper");
+  let childOne = document.getElementById("addWindow");
```
Comment (stezh) [inline src/main.js]: Better to change "childOne" to "addWindowFrame".

File: src/main.js [hunk 5]
```diff
@@ -699,14 +700,64 @@ function getNodePosition(node){
...
+  let childTwo = document.getElementById("addNoteBtn");
```
Comment (stezh) [inline src/main.js]: Better to change "childTwo" to "noteBtn".

File: src/main.js [hunk 6]
```diff
@@ -699,14 +700,64 @@ function getNodePosition(node){
...
+  let childFour = document.getElementById("addPicBtn");
```
Comment (stezh) [inline src/main.js]: Same as above.

File: src/main.js [hunk 7]
```diff
@@ -699,14 +700,64 @@ function getNodePosition(node){
...
+  let childThree = document.getElementById("closeWindowBtn");
```
Comment (stezh) [inline src/main.js]: Same as above.

File: src/main.js [hunk 8]
```diff
@@ -699,14 +700,64 @@ function getNodePosition(node){
...
+  childTwo.parentNode.removeChild(childTwo);
+  childFour.parentNode.removeChild(childFour);
+  childThree.parentNode.removeChild(childThree);
+  wrapper.parentNode.removeChild(wrapper);
+  addWindowOpen = false;
+}
 
-  var appendElement = '<div class="note" contenteditable="true" style="position: absolute;left: 8px;top: 8px;width: 200px;min-height: 200px;padding: 16px;box-shadow: 5px 5px 10px gray;background-color: rgb(255, 255, 150);font-size: 24pt;word-wrap: break-word;"></div>'
+function appendNote() {
+  let appendElement = '<div class="note" contenteditable="true" style="left: 8px;top: 8px;width: 235px;min-height: 100px;padding: 16px;box-shadow: 5px 5px 10px gray;background-color: rgb(255, 255, 150);font-size: 12pt;word-wrap: break-word;"></div><br>';
```
Comment (stezh) [inline src/main.js]: Define the static file inside the css file. 
I think "textarea" is better than contenteditable.

File: src/main.js [hunk 9]
```diff
@@ -699,14 +700,64 @@ function getNodePosition(node){
...
+function appendNote() {
+  let appendElement = '<div class="note" contenteditable="true" style="left: 8px;top: 8px;width: 235px;min-height: 100px;padding: 16px;box-shadow: 5px 5px 10px gray;background-color: rgb(255, 255, 150);font-size: 12pt;word-wrap: break-word;"></div><br>';    
+  let addWindow = document.getElementById("addWindow");
+  addWindow.contentWindow.document.body.innerHTML += appendElement;
+}
 
-  // Don't know why the elements does not append. TODO
-  var openWindow = window.open(newNodeAddress).document.body.innerHTML += appendElement;
+function appendPic() {
+  let appendElement = '<img src="" width="235" alt="Loading Image..."><br>';
```
Comment (stezh) [inline src/main.js]: use D3 to add element.

File: src/main.js [hunk 10]
```diff
@@ -722,15 +773,13 @@ function addEleContent(e) {
...
       console.warn("Node/Link NOT FOUND");
   }
-
-
 }
 
 function showContent(ele) {
-    var addressToFrame = "http://webstrates.ucsd.edu/" + ele;
-    var wrapper = document.createElement('div');
+    let addressToFrame = "http://webstrates.ucsd.edu/" + ele;
```
Comment (stezh) [inline src/main.js]: Create a new constant in constant.js to represent WEBSTRATES_URL_PREFIX.

File: src/main.js [hunk 11]
```diff
@@ -722,15 +773,13 @@ function addEleContent(e) {
...
+    let wrapper = document.createElement('div');
     wrapper.setAttribute("id", "showing");
-    var toFrame = '<iframe src="'+ addressToFrame + '" style="position: absolute;left: 8px;top: 8px;width: 300px;height: 300px;"><p>ERROR: Your browser does not support iframes.</p></iframe>';
+    let toFrame = '<iframe src="'+ addressToFrame + '" style="position: absolute;right: 8px;bottom: 8px;width: 300px;height: 300px;"><p>ERROR: Your browser does not support iframes.</p></iframe>';
```
Comment (stezh) [inline src/main.js]: Use JS to add element and change style in css.

File: index.html (hunk 1 above)
Comment (XavierXinweiWang) [reply index.html]: Left Undone For future work.

File: index.html (hunk 2 above)
Comment (XavierXinweiWang) [reply index.html]: Left Undone For future work.

File: src/main.js (hunk 3 above)
Comment (XavierXinweiWang) [reply src/main.js]: Refactored yet break AddPic() for unknown Reason, refactored code commented out for future exploration.

File: src/main.js (hunk 4 above)
Comment (XavierXinweiWang) [reply src/main.js]: Done

File: src/main.js (hunk 5 above)
Comment (XavierXinweiWang) [reply src/main.js]: Done

File: src/main.js (hunk 6 above)
Comment (XavierXinweiWang) [reply src/main.js]: Done

File: src/main.js (hunk 7 above)
Comment (XavierXinweiWang) [reply src/main.js]: Done

File: src/main.js (hunk 8 above)
Comment (XavierXinweiWang) [reply src/main.js]: Done

File: src/main.js (hunk 9 above)
Comment (XavierXinweiWang) [reply src/main.js]: Done (not using D3 though)

File: src/main.js (hunk 10 above)
Comment (XavierXinweiWang) [reply src/main.js]: Done

File: src/main.js (hunk 11 above)
Comment (XavierXinweiWang) [reply src/main.js]: Done

**Pull request merged**